lib/spark.js

```diff
@@ -71,11 +72,25 @@ export default class Spark {
     }
 
     console.log(`Querying IPNI to find retrieval providers for ${retrieval.cid}`)
-    const { indexerResult, provider } = await queryTheIndex(retrieval.cid, stats.providerId)
+    const { indexerResult, provider, providers } = await queryTheIndex(
+      retrieval.cid,
+      stats.providerId,
+    )
     stats.indexerResult = indexerResult
 
     const providerFound = indexerResult === 'OK' || indexerResult === 'HTTP_NOT_ADVERTISED'
-    if (!providerFound) return
+    const noValidAdvertisement = indexerResult === 'NO_VALID_ADVERTISEMENT'
+
+    // In case index lookup failed due to network error or CID not found,
+    // we will not perform any retrieval
```

Comment:
```suggestion
    // In case index lookup failed we will not perform any retrieval
```

Don't want to have to maintain this comment, in case the reasons for index lookup failure change